docs/dev-guide/contributing/dev_setup.rst

```diff
@@ -5,12 +5,90 @@ Developer Setup
 ===============
 
 For the Impatient
-^^^^^^^^^^^^^^^^^
+-----------------
 
-These instructions will create a developer install of Pulp on a dedicated
-development instance.
+There are two ways to automatically configure a development environment. There is a Vagrantfile
+in the platform git repository that can automatically deploy a virtual machine on your host with a
+Pulp development environment configured. Alternatively, there is a script that can turn a blank
+running virtual machine into a Pulp development environment.
 
-* Start a RHEL 7 or Fedora 20/21 x86_64 instance that will be dedicated for development with
+Vagrant
+^^^^^^^
+
+`Vagrant <https://docs.vagrantup.com/>`_ is a tool to aid developers in quickly deploying
+development environments. Pulp has provided a ``Vagrantfile`` in the platform git repository. This
+is the easiest way to get started on developing with Pulp if you aren't sure which method you
+prefer. Vagrant is available in Fedora 21 and newer. Follow these steps:
+
+#. Install vagrant::
+   
+   $ sudo yum install vagrant-libvirt
+
+#. Install NFS. NFS will be used to share your code directory with the deployed virtual machine::
+   
+   $ sudo yum install nfs-utils
+
+#. You will need to grant the nfsnobody user rx access to the folder that you check out your code
+   under. Many developers check out code into $HOME/devel or similar. In Fedora, $HOME typically
+   does not allow such access to other users. If your code is in $HOME, you will need to::
+   
+   $ setfacl -m user:nfsnobody:r-x $HOME  # Season to taste, as per above
```

Comment:
If your code lives in a dir under $HOME, you will need to setfacl on both $HOME and the dir underneath.